py-polars/tests/unit/io/test_pyarrow_dataset.py

```diff
@@ -199,3 +202,34 @@ def test_pyarrow_dataset_comm_subplan_elim(tmp_path: Path) -> None:
     assert lf0.join(lf1, on="a", how="inner").collect().to_dict(as_series=False) == {
         "a": [1, 2]
     }
+
+
+def test_pyarrow_dataset_with_pyarrow_options(tmp_path: Path) -> None:
+    df = pl.DataFrame({"a": [1, 2, 3]})
+
+    file_path_0 = tmp_path / "0.parquet"
+
+    df.write_parquet(file_path_0)
+
+    pa_ds = ds.dataset(file_path_0, format="parquet")
+    lf = pl.scan_pyarrow_dataset(pa_ds, pyarrow_options={"use_threads": False})
```

Comment:
It's better to use an option that we can actually see the effect of. I'd propose using a two-column dataframe and passing `columns` here (in addition to `use_threads`).